src/soc/ibm/power9/rs4.c

```diff
@@ -0,0 +1,828 @@
+/* SPDX-License-Identifier: Apache-2.0 */
```

Comment:
Maybe add a comment that this was copied without modifications with a link to source?